frame/fee-market/src/s2s/payment.rs

```diff
@@ -188,61 +212,35 @@ where
 	}
 }
 
-/// Slash order assigned relayers
-pub fn slash_assigned_relayers<T: Config>(
-	order: Order<T::AccountId, T::BlockNumber, RingBalance<T>>,
-	relayer_fund_account: &T::AccountId,
-) -> RingBalance<T> {
-	let mut total_slash = RingBalance::<T>::zero();
-	match (order.confirm_time, order.range_end()) {
-		(Some(confirm_time), Some(end_time)) if confirm_time >= end_time => {
-			let timeout = confirm_time - end_time;
-			let message_fee = order.first_and_last_fee().1.unwrap_or_default();
-			let slash_max = T::Slasher::slash(message_fee, timeout);
-			debug_assert!(
-				slash_max <= T::MiniumLockCollateral::get(),
-				"The maximum slash value returned from Slasher is MiniumLockCollateral"
-			);
-
-			for assigned_relayer in order.relayers_slice() {
-				let slashed_asset =
-					do_slash::<T>(&assigned_relayer.id, relayer_fund_account, slash_max);
-				total_slash += slashed_asset;
-			}
-		}
-		_ => {}
-	}
-	total_slash
-}
-
 /// Do slash for absent assigned relayers
 pub fn do_slash<T: Config>(
 	slash_account: &T::AccountId,
 	fund_account: &T::AccountId,
-	slash_max: RingBalance<T>,
+	slash_value: RingBalance<T>,
 ) -> RingBalance<T> {
-	let slashed;
-	let locked_collateral = crate::Pallet::<T>::relayer_locked_collateral(&slash_account);
+	let locked_collateral = crate::Pallet::<T>::relayer(&slash_account).collateral;
 	T::RingCurrency::remove_lock(T::LockId::get(), &slash_account);
-	if locked_collateral >= slash_max {
-		slashed = slash_max;
-		let locked_reserved = locked_collateral.saturating_sub(slashed);
-		transfer_and_print_logs_on_error::<T>(slash_account, fund_account, slashed);
-		crate::Pallet::<T>::update_collateral(&slash_account, locked_reserved);
-	} else {
-		slashed = locked_collateral;
-		transfer_and_print_logs_on_error::<T>(slash_account, fund_account, slashed);
-		crate::Pallet::<T>::update_collateral(&slash_account, RingBalance::<T>::zero());
-	}
-	slashed
+	debug_assert!(
+		locked_collateral >= slash_value,
+		"The locked collateral must alway greater than slash max"
+	);
+
+	let _ = <T as Config>::RingCurrency::transfer(
+		slash_account,
+		fund_account,
+		slash_value,
+		ExistenceRequirement::AllowDeath,
+	);
```

Comment:
should we compare the slash value and the left balance, and choose the minimize of them?

Reply:
The design make sure that there are enough balance locked for slash, otherwise it it bug, so printing log might be enough for catch that bug.